apps/docs/vite.config.ts

```diff
@@ -0,0 +1,57 @@
+/* eslint-disable import/named */
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/* eslint-disable @typescript-eslint/naming-convention */
+import { defineConfig, loadEnv, UserConfigExport } from 'vite';
+
+interface IEnvVars {
+	[x: string]: any;
+}
+
+/**
+ * DefaultConfig.
+ *
+ * @param props - Options.
+ * @param props.mode - Mode.
+ * @returns Config.
+ */
+const defaultConfig = ({ mode }): UserConfigExport => {
+	const minify =
+		mode === 'development' || mode === 'storybook' ? false : 'terser';
+	const sourcempas =
+		mode === 'development' || mode === 'storybook' ? true : false;
+
+	const ENV_VARS: IEnvVars = {
+		NODE_ENV: mode,
+		DEBUG: mode === 'development' || mode === 'storybook' ? true : false,
+		// eslint-disable-next-line no-undef
+		...process.env,
+		// eslint-disable-next-line no-undef
+		...loadEnv(mode, process.cwd()),
+	};
```

Comment:
Eslint rules were adjusted to avoid complains on not typed constants, therefore, IEnvVars was removed. 
 
Merge process.env vars, with loadEnv depending on the enviorment :

This implementation is not yet used in the project, nevertheless it is recommended to implement this additional control over the ENV vars due Vite restrictions on env variables naming convention. 
This feature is optional at this development stage, and can be resolved to be removed.

Reply:
This is not yet used in this project, But there are several Storybook plugins that requires ENV vars to be passed to storybook, that we have faced in other projects not to work with VITE restrictions on ENV naming conventions.